### ISSUE TITLE
Custom POST/PUT/DELETE subresourceOperations

### ISSUE THREAD

**Issue by Prometee** [api-platform/core#1558]
Hi,

I'm building an API with implicit relations between an Entity(in MySQL) and Entities(in SQLite).
So I try to build subresources with custom Operations because I need to connect to the right SQLite DB before getting all my sub-entities.

The goal at the end is to merge all SQLite Operations url with the URL of the "get" entityOperation of an Mysql Entity.

Actual Operations exemple :
```
Mysql Entity :
[GET] /entities
[POST] /entities
[GET] /entity/{id}
[PUT] /entity/{id}
[DELETE] /entity/{id}

SQLite Entity :
[GET] /entity/{id}/sqlite_entities #subresourceOperations give me only that possibility
[POST] /sqlite_entities
[GET] /sqlite_entity/{id}
[PUT] /sqlite_entity/{id}
[DELETE] /sqlite_entity/{id}
```
Expected Operations :
```
Mysql Entity :
[GET] /entities
[POST] /entities
[GET] /entity/{id}
[PUT] /entity/{id}
[DELETE] /entity/{id}

SQLite Entity :
[GET] /entity/{id}/sqlite_entities
[POST] /entity/{id}/sqlite_entities
[GET] /entity/{id}/sqlite_entity/{id}
[PUT] /entity/{id}/sqlite_entity/{id}
[DELETE] /entity/{id}/sqlite_entity/{id}
```

Is there a way to achieve this ? I already manage to override the subresourceOperation to list my SQLite SubRessources by connecting to the right DB, but I don't see how to achieve the same thing for put/post/delete and get entity operations by taking advantage of the (entity|collection)Operations already available for all SQLite entities I have.

But maybe the simple solution is to suffix all SQLite Operations path like : /sqlite_entity/{id}/{parent_entity_id}, but I would like a simple solution that only let me build the relation between those entities (Mysql/SQLite), it would be perfect :D

**Comment by Prometee** [comment]
I finally ended up with this :
 * decorating the `ResourceMetadataFactory` to add my custom route prefix (`/entity/{entity_id}`) only for SQLite entities, with just a single particularity to get Iri `"@Id"` working : a route defaults parameter containing `array("entity_id"=>0)`. Because `IriConverter::getIriFromResourceClass` don't give any $parameters to the `UrlGenerator` (It could be cool if a specific $context item could be given here) - I did not found any other better solution...
 * and adding a `DataProvider` which listen to the tags : 'api_platform.item_data_provider' and 'api_platform.collection_data_provider'. This service has `RequestStack` arguments to get the "entity_id" request parameter, then connect to the second `EntityManager->Connection` then throwing `ResourceClassNotSupportedException` to continue the normal execution of collections and entity Operations.

If someone understand what I explain :) Maybe it could help someone one day !

**Comment by Prometee** [comment]
I finally ended up with this :
 * decorating the `ResourceMetadataFactory` to add my custom route prefix (`/entity/{entity_id}`) only for SQLite entities, with just a single particularity to get Iri `"@Id"` working : a route defaults parameter containing `array("entity_id"=>0)`. Because `IriConverter::getIriFromResourceClass` don't give any $parameters to the `UrlGenerator` (It could be cool if a specific $context item could be given here) - I did not found any other better solution...
 * and adding a `DataProvider` which listen to the tags : 'api_platform.item_data_provider' and 'api_platform.collection_data_provider'. This service has `RequestStack` arguments to get the "entity_id" request parameter, then connect to the second `EntityManager->Connection` then throwing `ResourceClassNotSupportedException` to continue the normal execution of collections and entity Operations.

If someone understand what I explain :) Maybe it could help someone one day !
